vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/tracing/QueryRequest.java

```diff
@@ -0,0 +1,29 @@
+package io.vertx.sqlclient.impl.tracing;
+
+import io.vertx.sqlclient.Tuple;
+
+import java.util.List;
+
+/**
+ * A traceable query.
+ */
+public class QueryRequest {
+
+  final SqlTracer tracer;
+  final String sql;
+  final List<Tuple> tuples;
+
+  public QueryRequest(SqlTracer tracer, String sql, List<Tuple> tuples) {
+    this.tracer = tracer;
+    this.sql = sql;
+    this.tuples = tuples;
```

Comment:
aren't tuples the prepared statement parameters ?

Reply:
I mean we could make sending statement parameters as part of payload an **option** for users. Because the parameter might be a big BLOB or text value, sending them over the network would  not be a good choice and there's hardly any benefit on this.

Reply:
this actually is not logged by the tracers, if you look at it, a tracer would need to cast the request to a QueryRequest and do something with parameters. It seems that by default tracers don't care about parameters. So we only provide opportunity for a tracer to downcast to a `QueryRequest` and do something with them if they way. If that's a blob then it is up to the tracer to care about the payload issue, not the SQL client

Reply:
ok I see how this works in `vertx-tracing`